packages/astro/src/core/ssr/css.ts

```diff
@@ -30,8 +28,9 @@ export function getStylesForURL(filePath: URL, viteServer: vite.ViteDevServer):
         css.add(importedModule.url); // note: return `url`s for HTML (not .id, which will break Windows)
       }
       crawlCSS(importedModule.id, scanned);
-      scanned.add(importedModule.id);
     }
+
+    scanned.add(moduleName.id);
```

Comment:
This change is just cleanup (we don’t need to try to add modules multiple times to `scanned`; also wait until scanned is done to skip it); not affecting the PR.